packages/theme-core-plugin--pattern-lab--php/pattern-lab.js

```diff
@@ -15,33 +16,50 @@ module.exports = (userConfig) => {
   );
   const plRoot = path.join(config.configFile, '../..');
   const plSource = path.join(plRoot, plConfig.sourceDir);
-  // const plPublic = path.join(plRoot, plConfig.publicDir);
+  const plPublic = path.join(plRoot, plConfig.publicDir);
   const consolePath = path.join(plRoot, 'core/console');
 
   function plBuild(done, errorShouldExit) {
-    core.utils.sh(`php ${consolePath} --generate`, errorShouldExit, (err) => {
+    core.utils.sh(`php ${consolePath} --generate --patternsonly`, errorShouldExit, (err) => {
       core.events.emit('reload');
       done(err);
     });
   }
 
+
   function compile(done) {
     plBuild(done, true);
   }
   compile.description = 'Compile Pattern Lab';
 
+
+  function clean(done) {
+    del([
+      `${plPublic}/patterns/**/*`
+    ], { force: true }).then(() => done());
+  }
+  clean.description = 'Clean up markup generated by Pattern Lab';
+
+
   function watch() {
     const watchedExtensions = config.watchedExtensions.join(',');
     const plGlob = [path.normalize(`${plSource}/**/*.{${watchedExtensions}}`)];
     const src = config.extraWatches
       ? [].concat(plGlob, config.extraWatches)
       : plGlob;
-    gulp.watch(src, done => plBuild(done, false));
+
+    gulp.watch(src, (done) => {
+      clean(() => {
+        plBuild(done, false);
+      });
+    });
```

Comment:
I'm ok doing a clean before the build and taking that responsibility away from PL core, but the current output from this task when a file is changes looks like this:

```
[13:20:12] Starting '<anonymous>'...
configuring pattern lab...
your site has been generated...
site generation took 0.099728107452393 seconds and used 6MB of memory...
klaatu barada nikto...
[13:20:12] Finished '<anonymous>' after 205 ms
```

I'd like the name to be informative. Naming the function does the trick:

```js
    gulp.watch(src, function pl(done) {
      clean(() => {
        plBuild(done, false);
      });
    });
```

Also, I don't think we should export the `clean` function. It's not truly a clean function as you can't really delete the `public` directory without having to set it up again.